### ISSUE TITLE
DocC plugin injection taints `versions.resolved_dependencies`

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#1979]
We're inadvertently picking up the injected DocC plugin dependency when we're reporting resolved packages:

<img width="571" alt="CleanShot 2022-08-18 at 11 02 18@2x" src="https://user-images.githubusercontent.com/65520/185355937-b3a21f53-95d9-40ec-b65f-914d2d696f65.png">

We need to either capture resolved dependencies before that happens or revert the changes to the manifest(s).

**Comment by finestructure** [comment]
Fixed by https://gitlab.com/finestructure/swiftpackageindex-builder/-/merge_requests/140

**Comment by finestructure** [comment]
This is proving to be a bit trickier than expected.

The first stab didn't work, because reverting the `Package.swift` changes before reporting isn't enough. We'd also have to re-run package resolution, which can be expensive. So instead I changed it to capture `Package.resolved` right after the build step.

That's when I discovered, that SPM doesn't actually create a `Package.resolved` when a package has no dependencies. A better way to determine dependencies would be to use `swift package show-dependencies --format json`. However, that's a bit of a bigger change _and_ it also triggers package resolution. So it's not something to do lightly.

`DependencyResolution` is a module which lives in the server project and is used by both the server and the builder and it currently assumes to be run as a cheap operation looking for a `Package.resolved` file. We shouldn't change that.

The only "problem" with `getResolvedDependencies(...) -> [ResolvedDependency]?` is that so far we've assumed a return value of `nil` to mean "don't know" when there's no `Package.resolved` file. This is obviously distinct from "no dependencies". The difference is really whether you run this immediately after having run package resolution. If that's the case, then "no Package.resolved" really means "no dependencies".

I think the best way to deal with this is to leave `getResolvedDependencies` as is. When we run it in the server context there's no package resolution and `nil` keeps its semantic of "don't know".

In the builder, we run `getResolvedDependencies` right after build (i.e. package resolution) and therefore we can interpret `nil` as `[]`.
